server/src/main/java/org/apache/druid/server/coordinator/DataSourceCompactionConfig.java

```diff
@@ -104,6 +100,43 @@ public DataSourceCompactionConfig(
     );
   }
 
+  /**
+   * Copied from CompactionTask#getValidTargetCompactionSizeBytes.
+   * Must be synced.
+   */
+  @Nullable
+  private static Long getValidTargetCompactionSizeBytes(
+      @Nullable Long targetCompactionSizeBytes,
+      @Nullable Integer maxRowsPerSegment,
+      @Nullable UserCompactTuningConfig tuningConfig
+  )
+  {
+    if (targetCompactionSizeBytes != null) {
+      Preconditions.checkArgument(
+          !hasPartitionConfig(maxRowsPerSegment, tuningConfig),
+          "targetCompactionSizeBytes[%s] cannot be used with maxRowsPerSegment[%s] and maxTotalRows[%s]",
+          targetCompactionSizeBytes,
+          maxRowsPerSegment,
+          tuningConfig == null ? null : tuningConfig.getMaxTotalRows()
+      );
+      return targetCompactionSizeBytes;
+    } else {
+      return hasPartitionConfig(maxRowsPerSegment, tuningConfig) ? null : DEFAULT_TARGET_COMPACTION_SIZE_BYTES;
+    }
+  }
+
+  /**
+   * Copied from CompactionTask#hasPartitionConfig.
+   * Must be synced.
```

Comment:
Can you add a comment here and elsewhere explaining why it's copied and has to be synced?

Reply:
Thanks, added.